src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -142,7 +142,7 @@ private function addAccessControlSection(ArrayNodeDefinition $rootNode)
                             ->scalarNode('host')->defaultNull()->end()
                             ->integerNode('port')->defaultNull()->end()
                             ->arrayNode('ips')
-                                ->beforeNormalization()->ifString()->then(function ($v) { return [$v]; })->end()
+                                ->beforeNormalization()->ifString()->then(function ($v) { return preg_split('/\s*,\s*/', $v); })->end()
```

Comment:
this change is not needed as it won't do anything with env vars

Reply:
This make able to validate ips list in \Symfony\Bundle\SecurityBundle\DependencyInjection\SecurityExtension::createRequestMatcher with isValidIp method (884 line).

Reply:
my point is that this won't work with env vars, since by definition their values are unknown at compile time, so we cannot validation anything when this logic runs.

Reply:
Yes, this change had made only for comma separated ips, passed as static string.

Reply:
Why should we allow this syntax? I'm not sure we should actually.

Reply:
I think if we allow to pass comma separated ips via env var, we should allow do it via simple string for consistence.

Reply:
That's not what this line allows actually: it allows for the array of ios to be given as a string.
But it doesn't allow strings *inside the array* to be given as csv, while this is precisely what would be allowed for envvars.
For static strii, I don't think it's a good idea to provide two syntax when not required.

Reply:
Actually csv or json strings cannot be used there.
If you'll try to pass `%env(csv:VAR)%` (or `%env(json:VAR)%`) as value for `ips`, you'll get InvalidTypeException from \Symfony\Component\Config\Definition\BaseNode::doValidateType because array node doesn't allow to use dynamic values to configure itself.
If you'll try to pass that env var as value in array, you'll get InvalidTypeException because csv/json resolves to array, not scalar.

Reply:
I was talking about a static csv string, as you did also before.

Reply:
Maybe you'll get my point with an example.

Your patch on this line allows this:
```yaml
ips: 'ip1, ip2'
```

I think we should not allow it, especially when `ips: [ip1, ip2]` works already.

The other part of your patch allows this:
```yaml
ips: ['%env(IPS)%']
```

And it allows the env var to be a csv list of ips.

But this is not allowed:
```yaml
ips: ['ip1, ip2']
```

Following your logic, that's what you were actually looking to allow "for consistency".

But I also don't think it would be a good idea, because there is already a better supported syntax.

Reply:
Now I'm understood what you mean.